crates/oxc_linter/src/rules/vitest/prefer_spy_on.rs

```diff
@@ -0,0 +1,306 @@
+use oxc_ast::{
+    AstKind,
+    ast::{
+        Argument, AssignmentExpression, CallExpression, Expression, MemberExpression,
+        SimpleAssignmentTarget,
+    },
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_semantic::AstNode;
+use oxc_span::Span;
+
+use crate::{
+    context::LintContext,
+    fixer::RuleFixer,
+    rule::Rule,
+    utils::{PossibleJestNode, get_node_name, parse_general_jest_fn_call},
+};
+
+fn use_vi_spy_one(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Suggest using `vi.spyOn()`.")
+        .with_help("Use vi.spyOn() instead")
+        .with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct PreferSpyOn;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// When mocking a function by overwriting a property you have to manually restore
+    /// the original implementation when cleaning up. When using `vi.spyOn()` Vitest
+    /// keeps track of changes, and they can be restored with `vi.restoreAllMocks()`,
+    /// `mockFn.mockRestore()` or by setting `restoreMocks` to `true` in the Vitest
+    /// config.
+    ///
+    /// Note: The mock created by `vi.spyOn()` still behaves the same as the original
+    /// function. The original function can be overwritten with
+    /// `mockFn.mockImplementation()` or by some of the
+    /// [other mock functions](https://vitest.dev/api/mock.html).
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Directly overwriting properties with mock functions can lead to cleanup issues
+    /// and test isolation problems. When you manually assign a mock to a property,
+    /// you're responsible for restoring the original implementation, which is easy to
+    /// forget and can cause tests to interfere with each other. Using `vi.spyOn()`
+    /// provides automatic cleanup capabilities and makes your tests more reliable.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```javascript
+    /// Date.now = vi.fn();
+    /// Date.now = vi.fn(() => 10);
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```javascript
+    /// vi.spyOn(Date, 'now');
+    /// vi.spyOn(Date, 'now').mockImplementation(() => 10);
+    /// ```
+    PreferSpyOn,
+    vitest,
+    style,
+    fix,
+);
+
+impl Rule for PreferSpyOn {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::AssignmentExpression(assign_expr) = node.kind() else {
+            return;
+        };
+
+        let left = &assign_expr.left;
+        let right = &assign_expr.right;
+
+        let Some(left_assign) = left
+            .as_simple_assignment_target()
+            .and_then(SimpleAssignmentTarget::as_member_expression)
+        else {
+            return;
+        };
+
+        match right {
```

Comment:
These could be inlined since they're only used once:


```suggestion
        let Some(left_assign) = &assign_expr.left
            .as_simple_assignment_target()
            .and_then(SimpleAssignmentTarget::as_member_expression)
        else {
            return;
        };

        match &assign_expr.right {
```